test/pylib/pool.py

```diff
@@ -97,7 +97,7 @@ async def __aenter__(self):
                 return self.obj
 
             async def __aexit__(self, exc_type, exc, obj):
-                if self.obj:
+                if exc_type is None and self.obj:
```

Comment:
The entire point of context managers is to release resources even in presence of exceptions. This makes this context manager act in a confusing way, unlike a context manager should.

We should stop using a context manager if we want to introduce complex if-logic for releasing or not releasing resources at the end.

Reply:
What do you think of a  `Pool.destroy()` symmetric to `build()`.